templates/articles/list.html

```diff
@@ -1,28 +1,76 @@
-{% for article in articles %}
-<div class="article-preview">
-  <div class="article-meta">
-    <a href="{% url 'profile_view' profile=article.author.user.username %}"
-      ><img src="{{ article.author.img }}"
-    /></a>
-    <div class="info">
+<turbo-frame id="article-list">
+  {% for article in page_obj %}
+  <div class="article-preview">
+    <div class="article-meta">
       <a
         href="{% url 'profile_view' profile=article.author.user.username %}"
-        class="author"
-      >
-        {{ article.author.user.username }}
-      </a>
-      <span class="date">{{ article.created_at }}</span>
+        target="_top"
+        ><img src="{{ article.author.img }}"
+      /></a>
+      <div class="info">
+        <a
+          href="{% url 'profile_view' profile=article.author.user.username %}"
+          class="author"
+          target="_top"
+        >
+          {{ article.author.user.username }}
+        </a>
+        <span class="date">{{ article.created_at }}</span>
+      </div>
+      <button class="btn btn-outline-primary btn-sm pull-xs-right">
+        <i class="ion-heart"></i> {{ article.favorited_by__count }}
+      </button>
     </div>
-    <button class="btn btn-outline-primary btn-sm pull-xs-right">
-      <i class="ion-heart"></i> {{ article.favorited_by__count }}
-    </button>
+    <a
+      href="{% url 'article_view' article.slug %}"
+      class="preview-link"
+      target="_top"
+    >
+      <h1>{{ article.title }}</h1>
+      <p>{{ article.description }}</p>
+      <span>Read more...</span>
+    </a>
+  </div>
+  {% empty %}
+  <div class="article-preview">No articles are here... yet.</div>
+  {% endfor %}
+
+  <div class="pagination">
+    <span class="step-links">
+      {% if page_obj.has_previous %}
+      <a
+        onClick="toTop()"
+        href="{% url 'article_list' %}?page=1&{{query_string}}"
+        >&laquo; first</a
+      >
+      <a
+        onClick="toTop()"
+        href="{% url 'article_list' %}?page={{ page_obj.previous_page_number }}&{{query_string}}"
+        >previous</a
+      >
+      {% endif %}
+
+      <span class="current">
+        Page {{ page_obj.number }} of {{ page_obj.paginator.num_pages }}.
+      </span>
+
+      {% if page_obj.has_next %}
+      <a
+        onClick="toTop()"
+        href="{% url 'article_list' %}?page={{ page_obj.next_page_number }}&{{query_string}}"
+        >next</a
+      >
+      <a
+        onClick="toTop()"
+        href="{% url 'article_list' %}?page={{ page_obj.paginator.num_pages }}&{{query_string}}"
+        >last &raquo;</a
+      >
+      {% endif %}
+    </span>
```

Comment:
Added pagination controls. This is just lifted from the Django docs. Not super pretty, but works. `query_string` is used to preserve the current filtering while moving through the pages of results. Not sure if there's a more elegant way to persist this state.